### ISSUE TITLE
cell-var-from-loop misses case with nested loop

### ISSUE THREAD

**Issue by andy-ms** [PyCQA/pylint#2846]
### Steps to reproduce
1. Create a file  `a.py` containing:
```py
TEST_A = [
    (lambda: n)
    for n in range(3)
]

TEST_B = [
    (lambda: [n for _ in range(3)])
    for n in range(3)
]

print(TEST_A[1]())  # User probably intended for this to be 1, actually 2

print(TEST_B[1]())  # User probably intended for this to be [1, 1, 1], actually [2, 2, 2]
```
2. Run `pylint a.py`

### Current behavior

There is a lint failure: `a.py:2:13: W0640: Cell variable n defined in loop (cell-var-from-loop)`
Tere is no such failure for `TEST_B`.

### Expected behavior

Lint failures for both `TEST_A` and `TEST_B`.

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 22:20:52) [MSC v.1916 32 bit (Intel)]
```

**Comment by PCManticore** [comment]
Thanks for the report @andy-ms

**Comment by SamyCookie** [comment]
Does https://github.com/PyCQA/pylint/pull/2276 fixed this issue ?

**Comment by PCManticore** [comment]
The error still seems to be here on `master` branch.
